pkg/network/server.go

```diff
@@ -514,7 +514,7 @@ func (s *Server) run() {
 			if s.peers[drop.peer] {
 				delete(s.peers, drop.peer)
 				s.lock.Unlock()
-				if errors.Is(drop.reason, errInvalidInvType) || errors.Is(drop.reason, errStateMismatch) || strings.Contains(drop.reason.Error(), errInvalidBlock.Error()) {
+				if strings.Contains(drop.reason.Error(), errInvalidInvType.Error()) || errors.Is(drop.reason, errStateMismatch) || strings.Contains(drop.reason.Error(), errInvalidBlock.Error()) {
```

Comment:
But why?

Reply:
> errInvalidInvType should be extended: we need to format this error to include the inventory time itself. Implement it in a separate commit.

so I changed `errInvalidInvType` to `fmt.Errorf("%w: %s", errInvalidInvType, inv.Type.String())` and that why we need compare strings

Reply:
`errors.Is` should work with it.